br/tests/br_pitr/run.sh

```
@@ -23,182 +23,230 @@ PREFIX="pitr_backup" # NOTICE: don't start with 'br' because `restart services`
+    else
+        run_br --pd $PD_ADDR restore point -s "local://$TEST_DIR/$PREFIX/log" --start-ts $current_ts || restore_fail=1
+    fi
+    if [ $restore_fail -ne 1 ]; then
+        echo 'pitr success on non compliant operation'
+        exit 1
+    fi
+
+    # clean up failed restore registry entry
+    echo "clean up failed restore registry entry"
+    if [ "$USE_TABLE_FILTER" = "true" ]; then
+        run_br --pd $PD_ADDR abort restore point -s "local://$TEST_DIR/$PREFIX/log" --start-ts $current_ts -f "db_to_be_dropped*.*" -f "table_to_be_dropped_or_truncated*.*" -f "key_types_test.*" -f "br_pitr.*" -f "test.*" -f "partition_to_be_dropped_or_truncated*.*" -f "partition_to_be_removed_or_altered*.*" -f "index_or_primarykey_to_be_dropped*.*" -f "indexes_to_be_dropped*.*" -f "column_s_to_be_dropped*.*" -f "column_to_be_modified*.*" || restore_fail=1
```

Shellcheck (0.11.0)

[info] Double quote to prevent globbing and word splitting.

(SC2086)